### PR TITLE
Runtime clean up part 1 of many

### PR DESCRIPTION
**Pull request by davidfowl**
- Removed unused ExceptionHelper.cs
- Clean up unused dependencies
- Removed the use of Linq in the core pieces

### CODE REVIEW

File: src/kre.host/Bootstrapper.cs [hunk 1]
```diff
@@ -35,7 +35,8 @@ public Task<int> RunAsync(string[] args)
             try
             {
                 var name = args[0];
-                var programArgs = args.Skip(1).ToArray();
+                var programArgs = new string[args.Count - 1];
+                args.CopyTo(1, programArgs, 0, programArgs.Length);
```
Comment (NTaylorMullen) [inline src/kre.host/Bootstrapper.cs]: named params, especially since `CopYTo` isn't use very frequently.

File: src/kre.host/Bootstrapper.cs [hunk 2]
```diff
@@ -17,14 +17,14 @@ namespace kre.host
...
+        private readonly IEnumerable<string> _searchPaths;
 
-        public Bootstrapper(string[] searchPaths)
+        public Bootstrapper(IEnumerable<string> searchPaths)
         {
             _searchPaths = searchPaths;
         }
 
-        public Task<int> RunAsync(string[] args)
+        public Task<int> RunAsync(List<string> args)
```
Comment (NTaylorMullen) [inline src/kre.host/Bootstrapper.cs]: Need to be a list? would an IList sufice?

File: src/kre.host/RuntimeBootstrapper.cs [hunk 3]
```diff
@@ -131,13 +130,13 @@ public static Task<int> ExecuteAsync(string[] args)
             }
 
             // Resolve the lib paths
-            string[] searchPaths = ResolveSearchPaths(optionLib.Values, app.RemainingArguments);
+            IEnumerable<string> searchPaths = ResolveSearchPaths(optionLib.Values, app.RemainingArguments);
```
Comment (NTaylorMullen) [inline src/kre.host/RuntimeBootstrapper.cs]: var

File: src/kre.host/RuntimeBootstrapper.cs [hunk 4]
```diff
@@ -131,13 +130,13 @@ public static Task<int> ExecuteAsync(string[] args)
...
 
             var bootstrapper = new Bootstrapper(searchPaths);
-            return bootstrapper.RunAsync(app.RemainingArguments.ToArray());
+            return bootstrapper.RunAsync(app.RemainingArguments);
         }
 
-        private static string[] ResolveSearchPaths(IEnumerable<string> libPaths, List<string> remainingArgs)
+        private static IEnumerable<string> ResolveSearchPaths(List<string> libPaths, List<string> remainingArgs)
```
Comment (NTaylorMullen) [inline src/kre.host/RuntimeBootstrapper.cs]: IList?

File: src/kre.host/RuntimeBootstrapper.cs [hunk 5]
```diff
@@ -171,14 +173,18 @@ private static string[] ResolveSearchPaths(IEnumerable<string> libPaths, List<st
                 }
             }
 
-            return searchPaths.ToArray();
+            return searchPaths;
         }
 
-        private static IEnumerable<string> ExpandSearchPath(string libPath)
+        private static void ExpandSearchPath(string libPath, List<string> searchPaths)
```
Comment (NTaylorMullen) [inline src/kre.host/RuntimeBootstrapper.cs]: ditto.

File: src/kre.host/RuntimeBootstrapper.cs [hunk 6]
```diff
@@ -171,14 +173,18 @@ private static string[] ResolveSearchPaths(IEnumerable<string> libPaths, List<st
...
         {
-            // Expand ; separated arguments
-            return libPath.Split(_libPathSeparator, StringSplitOptions.RemoveEmptyEntries)
-                          .Select(Path.GetFullPath);
+            if (libPath.IndexOf(';') >= 0)
```
Comment (NTaylorMullen) [inline src/kre.host/RuntimeBootstrapper.cs]: Contains

File: src/kre.host/RuntimeBootstrapper.cs (hunk 4 above)
Comment (davidfowl) [inline src/kre.host/RuntimeBootstrapper.cs]: This method is private

File: src/kre.host/RuntimeBootstrapper.cs (hunk 6 above)
Comment (davidfowl) [inline src/kre.host/RuntimeBootstrapper.cs]: Contains takes a string

**Pull request merged**